pandas/tests/dtypes/cast/test_infer_dtype.py

```diff
@@ -153,8 +154,8 @@ def test_infer_dtype_from_scalar_errors():
         (np.array([[1.0, 2.0]]), np.float_, False),
         (Categorical(list("aabc")), np.object_, False),
         (Categorical([1, 2, 3]), np.int64, False),
-        (Categorical(list("aabc")), "category", True),
-        (Categorical([1, 2, 3]), "category", True),
+        (Categorical(list("aabc")), CategoricalDtype(categories=["a", "b", "c"]), True),
+        (Categorical([1, 2, 3]), CategoricalDtype(categories=[1, 2, 3]), True),
```

Comment:
oh we now return a CDT from infer_dtype_from_array and not a string?

Reply:
but this is not tru for DatetimeTZ though?

Reply:
hmm this is troublesome.

This doesn't change what infer_dtype_from_array returns, but it _does_ change how is_dtype_equal behaves:

```
from pandas.tests.dtypes.cast.test_infer_dtype import *
arr = Categorical(list("aabc"))
pandas_dtype = True
dtype, _ = infer_dtype_from_array(arr, pandas_dtype=pandas_dtype)

>>> dtype
CategoricalDtype(categories=['a', 'b', 'c'], ordered=False)

>>> dtype == "category"
True
>>> is_dtype_equal(dtype, "category")
False
```

Which is unintentional.  I'll see if i can tweak is_dtype_equal to avoid this.